packages/ra-keycloak/src/authProvider.ts

```diff
@@ -97,24 +97,42 @@ export const keycloakAuthProvider = (
     } = {}
 ): AuthProvider => ({
     async login() {
+        let redirectUri = `${window.location.origin}/auth-callback`;
+        if (options.loginRedirectUri) {
+            if (!options.loginRedirectUri.startsWith('http')) {
+                redirectUri = `${window.location.origin}${options.loginRedirectUri}`;
+            } else {
+                redirectUri = options.loginRedirectUri;
+            }
+        }
         await initKeyCloakClient(keycloakClient, options.initOptions);
         return keycloakClient.login({
-            redirectUri:
-                options.loginRedirectUri ??
-                `${window.location.origin}/auth-callback`,
+            redirectUri,
         });
     },
     async logout() {
+        let redirectUri = window.location.origin;
+        if (options.logoutRedirectUri) {
+            if (!options.logoutRedirectUri.startsWith('http')) {
+                redirectUri = `${window.location.origin}${options.logoutRedirectUri}`;
+            } else {
+                redirectUri = options.logoutRedirectUri;
+            }
+        }
         await initKeyCloakClient(keycloakClient, options.initOptions);
         return keycloakClient.logout({
-            redirectUri:
-                options.logoutRedirectUri ?? `${window.location.origin}/login`,
+            redirectUri,
         });
     },
     async checkError() {
         await initKeyCloakClient(keycloakClient, options.initOptions);
         await isAuthenticated(keycloakClient, options.authenticationTimeout);
         if (!keycloakClient.authenticated || !keycloakClient.token) {
+            // not authenticated: save the location that the user tried to access
+            localStorage.setItem(
+                PreviousLocationStorageKey,
+                window.location.href.replace(window.location.origin, '')
+            );
```

Comment:
Can't say if this was working before or not (can't remember if I tested it), but opening a URL like `http://localhost:8081/posts/13` for the first time doesn't redirect me there after filling in my credentials.

Happens both in Browser and Hash Router modes.

Reply:
Fixed